### PR TITLE
RSDK-3449: update getkinematics and getgeometries wrappers

### PR DESCRIPTION
**Pull request by purplenicole730**
[RSDK-3449 ticket](https://viam.atlassian.net/browse/RSDK-3449)

This is to get ready for the new automated proto update that introduced two new methods. The new common methods `GetKinematics` and `GetGeometries` are now added to the component wrappers where needed.
Service tests have also been added to confirm that a `MethodNotImplementedError` is raised.

The `base` component now also has a `GetProperties` function that gives the `width` and `turn radius`. The PR's scope was changed to include the inclusion of this change so that all errors from the new protos update were addressed.

### CODE REVIEW

**Review by jckras — APPROVED** [review]
LGTM

**Comment by benjirewis** [comment]
It looks like CI is failing because of issues with imports; any idea what's going on there?

File: src/viam/components/base/client.py [hunk 1]
```diff
@@ -107,13 +109,18 @@ async def stop(
         request = StopRequest(name=self.name, extra=dict_to_struct(extra))
         await self.client.Stop(request, timeout=timeout)
 
-    async def is_moving(self, *, extra: Optional[Dict[str, Any]] = None, timeout: Optional[float] = None) -> bool:
-        if extra is None:
-            extra = {}
+    async def is_moving(self, *, timeout: Optional[float] = None) -> bool:
```
Comment (benjirewis) [inline src/viam/components/base/client.py:112]: What happened to the `extra` parameter here?

File: src/viam/components/base/service.py [hunk 2]
```diff
@@ -111,3 +124,6 @@ async def DoCommand(self, stream: Stream[DoCommandRequest, DoCommandResponse]) -
         result = await base.do_command(command=struct_to_dict(request.command), timeout=timeout, metadata=stream.metadata)
         response = DoCommandResponse(result=dict_to_struct(result))
         await stream.send_message(response)
+
+    async def GetGeometries(self, stream: Stream[GetGeometriesRequest, GetGeometriesResponse]) -> None:
+        raise MethodNotImplementedError("GetGeometries").grpc_error
```
Comment (benjirewis) [inline src/viam/components/base/service.py:129]: Do all the services not need `GetKinematics`, too?

Comment (purplenicole730) [reply src/viam/components/base/service.py:129]: No, GetKinematics is only used by the `arm` component.

File: src/viam/components/base/client.py (hunk 1 above)
Comment (purplenicole730) [reply src/viam/components/base/client.py:112]: I noticed that `IsMovingRequest` doesn't actually take `extra`, so I deleted it.

Comment (benjirewis) [reply src/viam/components/base/client.py:112]: Ah interesting, yeah I thought all gRPC messages had an `extra` field for some reason, but looks like you have to explicitly define one.

Comment (benjirewis) [reply src/viam/components/base/client.py:112]: This is hypothetically backward-breaking though, what are your thoughts on removing this @njooma?

Comment (purplenicole730) [reply src/viam/components/base/client.py:112]: @benjirewis Naveed has agreed that removing it would be the way to go.

**Review by benjirewis — APPROVED** [review]
LGTM! Thanks

**Pull request merged**